File: SwiatZwierzat_C++/Giraffe.cpp
#pragma once
#include "Mammal.cpp"
#include "Herbivore.cpp"

class Giraffe : public Mammal, public Herbivore{  //*dziedziczenie* slowo public oznacza, że dziedziczone elementy (np. zmienne lub funkcje) mają taką widoczność jak w klasie bazowej, oprocz private w bazowej. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
				//mozemy stworzyc oddzielna klase, czyli np Mammal i odziedziczyc po niej zestaw cech które ma kazdy ssak

public:
    Giraffe() : Mammal() { //wywolujemy konstruktor klasy, po ktorej dziedziczymy, bo zyrafa tez jest ssakiem
		cout << "Jestem zyrafa" << endl;
    }

    void eatPlant() {
        cout << "Jestem roslinozerca, zywie sie liscmi drzew" << endl;
    }

    void presentYourself() {
        cout << "Mam na imie " << getName() << " oraz zyje tu na sawannie juz od " << getAge() << " lat" << endl;
        eatPlant();
    }
};

File: SwiatZwierzat_Java/Giraffe.java
public class Giraffe extends Mammal implements Herbivore{  //*dziedziczenie* extends uzywamy podczas dziedziczenia klas. Zapobiega to powielaniu tych samych cech obiektow. Dzieki temu mozemy
    //możemy stworzyć oddzielną klasę, czyli Animal i odziedziczyć po niej zestaw cech które ma każde zwierzę
    //z kolei implements ma podobne znaczenie do extends jednak w tym przypadku oznajmiamy ze de facto bedziemy wykorzystywac dany interfejs,
    //korzystac i zapisywac cialo metod z interfejsu
    Giraffe(){
        super(); //za pomoca slowa super w tym przypadku wywolujemy konstruktor klasy, po ktorej dziedziczymy
        System.out.println("Jestem zyrafa");
    }

    @Override
    public void eatPlant() {
        System.out.println("Jestem roslinozerca, zywie sie liscmi drzew");
    }

    @Override
    public void presentYourself() {
        System.out.println("Mam na imie " + getName() + " oraz zyje tu na sawannie juz od " + getAge() + " lat");
        eatPlant();
    }
}
